### PR TITLE
[A4988] Add UnitsNet.Angle and rework Rotate method

### PR DESCRIPTION
**Pull request by torbacz**
## Description

- Installed UnitsNet.nanoFramework.Angle package.
- Replaced Rotate method arguments
- Adjust sample project
- Removed "sample" section from readme (it wasn't up to date). Replaced with "Please check sample project.". Easier to maintain sample project in one file :)

## Motivation and Context

- Accepting multiple angle formats 
- Mentioned here: https://github.com/nanoframework/nanoFramework.IoT.Device/pull/468#pullrequestreview-1023040489

## How Has This Been Tested?

- Tested on ESP32 with TMC2208 in standalone mode (works the same as A4988) 
- VS2022 PRO with latest NF extension
- Tested UnitsNet.Angle.FromDegrees and UnitsNet.Angle.FromRadians and passing full rotation. 

## Types of changes

- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [ ] New feature (non-breaking change which adds functionality to code)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [x] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.
- [x] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

### CODE REVIEW

File: devices/A4988/A4988 .cs [hunk 1]
```diff
@@ -58,16 +59,16 @@ protected virtual void SleepBetweenSteps()
...
         {
-            if (degree == 0)
+            if (angle.Value == 0)
             {
                 return;
             }
 
-            _dirPin.Write(degree > 0 ? PinValue.High : PinValue.Low);
-            var degreeForStepsCalculation = degree < 0 ? -degree : degree;
+            _dirPin.Write(angle.Value > 0 ? PinValue.High : PinValue.Low);
```
Comment (Ellerbach) [inline devices/A4988/A4988 .cs:70]: angle.Value will check the value of the angle based on the way it's been created.
So better use ange.Degree instead.

File: devices/A4988/README.md [hunk 2]
```diff
@@ -22,21 +22,4 @@ STEP - connect to microcontroller pin
...
-    var direction = true;
-    while (true)
-    {
-        motor.Rotate(360, direction);
-        direction = !direction;
-    }
-}
-```
\ No newline at end of file
+Please check sample project.
```
Comment (Ellerbach) [inline devices/A4988/README.md:25]: I prefer you keep the code :-) Just copy/paste, fine. But this does appear in the nuget description. So people can copy/paste from there ;-)

**Review by Ellerbach — COMMENTED** [review]
Few little things then good to go! And thanks a lot for making the PR on UnitsNet as well!

File: devices/A4988/README.md (hunk 2 above)
Comment (torbacz) [reply devices/A4988/README.md]: done

**Comment by torbacz** [comment]
> Few little things then good to go! And thanks a lot for making the PR on UnitsNet as well!

Done. No problem with UnitsNet.

**Pull request merged**